src/app/ReadHandler.h

```diff
@@ -226,8 +228,9 @@ class ReadHandler : public Messaging::ExchangeDelegate
     NodeId mInitiatorNodeId                                  = kUndefinedNodeId;
     AttributePathExpandIterator mAttributePathExpandIterator = AttributePathExpandIterator(nullptr);
     bool mIsFabricFiltered                                   = false;
-    bool mHoldSync                                           = false;
-    uint32_t mLastWrittenEventsBytes                         = 0;
+    // only for subscription, mHoldSync means empty sync report can be generated after max interval,
+    bool mHoldSync                   = false;
```

Comment:
And if mHoldSync is false it meand the empty sync report cannot be generated?  That's what the comment is implying, but I don't think that's what it means to say....  Again, what happens if this is false?  What happens if it's true?

Reply:
mHoldSync is set to false after max interval is passed, which means empty sync report is allowed to be generated. mHoldSync is set to true when trying to refresh subscription timer, which mean empty sync report cannot be sent during max interval

Reply:
updated, thanks